From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Node.js GitHub Bot" <github-bot@iojs.org>
Date: Sun, 2 Aug 2026 06:56:37 -0400
Subject: [PATCH 48/60] crypto: update root certificates to NSS 3.125
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This is the certdata.txt[0] from NSS 3.125.

This is the version of NSS that shipped in Firefox 153.0 on 2026-07-21.

Certificates removed:
- Entrust Root Certification Authority
- SecureSign Root CA12

[0] https://raw.githubusercontent.com/nss-dev/nss/refs/tags/NSS_3_125_RTM/lib/ckfw/builtins/certdata.txt

PR-URL: https://github.com/nodejs/node/pull/64746
Reviewed-By: Antoine du Hamel <duhamelantoine1995@gmail.com>
Reviewed-By: Luigi Pinca <luigipinca@gmail.com>
Reviewed-By: Yagiz Nizipli <yagiz@nizipli.com>
Reviewed-By: Colin Ihrig <cjihrig@gmail.com>
Signed-off-by: Juan José Arboleda <soyjuanarbol@gmail.com>
(cherry picked from commit fe2a6b2be81775524b7753e1dd346a398da239fa)
---
 src/node_root_certs.h |  47 -----
 tools/certdata.txt    | 481 +++++++++++++++---------------------------
 2 files changed, 170 insertions(+), 358 deletions(-)

diff --git a/src/node_root_certs.h b/src/node_root_certs.h
index 53ac2034c8..48d2fc5cb7 100644
--- a/src/node_root_certs.h
+++ b/src/node_root_certs.h
@@ -1,31 +1,5 @@
 #if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
 
-/* Entrust Root Certification Authority */
-"-----BEGIN CERTIFICATE-----\n"
-"MIIEkTCCA3mgAwIBAgIERWtQVDANBgkqhkiG9w0BAQUFADCBsDELMAkGA1UEBhMCVVMxFjAU\n"
-"BgNVBAoTDUVudHJ1c3QsIEluYy4xOTA3BgNVBAsTMHd3dy5lbnRydXN0Lm5ldC9DUFMgaXMg\n"
-"aW5jb3Jwb3JhdGVkIGJ5IHJlZmVyZW5jZTEfMB0GA1UECxMWKGMpIDIwMDYgRW50cnVzdCwg\n"
-"SW5jLjEtMCsGA1UEAxMkRW50cnVzdCBSb290IENlcnRpZmljYXRpb24gQXV0aG9yaXR5MB4X\n"
-"DTA2MTEyNzIwMjM0MloXDTI2MTEyNzIwNTM0MlowgbAxCzAJBgNVBAYTAlVTMRYwFAYDVQQK\n"
-"Ew1FbnRydXN0LCBJbmMuMTkwNwYDVQQLEzB3d3cuZW50cnVzdC5uZXQvQ1BTIGlzIGluY29y\n"
-"cG9yYXRlZCBieSByZWZlcmVuY2UxHzAdBgNVBAsTFihjKSAyMDA2IEVudHJ1c3QsIEluYy4x\n"
-"LTArBgNVBAMTJEVudHJ1c3QgUm9vdCBDZXJ0aWZpY2F0aW9uIEF1dGhvcml0eTCCASIwDQYJ\n"
-"KoZIhvcNAQEBBQADggEPADCCAQoCggEBALaVtkNC+sZtKm9I35RMOVcF7sN5EUFoNu3s/poB\n"
-"j6E4KPz3EEZmLk0eGrEaTsbRwJWIsMn/MYszA9u3g3s+IIRe7bJWKKf44LlAcTfFy0cOlypo\n"
-"wCKVYhXbR9n10Cv/gkvJrT7eTNuQgFA/CYqEAOwwCj0Yzfv9KlmaI5UXLEWeH25DeW0MXJj+\n"
-"SKfFI0dcXv1u5x609mhF0YaDW6KKjbHjKYD+JXGIrb68j6xSlkuqUY3kEzEZ6E5Nn9uss2rV\n"
-"vDlUccp6en+Q3X0dgNmBu1kmwhH+5pPi94DkZfs0Nw4pgHBNrziGLp5/V6+eF67rHMsoIV+2\n"
-"HNjnogQi+dPa2MsCAwEAAaOBsDCBrTAOBgNVHQ8BAf8EBAMCAQYwDwYDVR0TAQH/BAUwAwEB\n"
-"/zArBgNVHRAEJDAigA8yMDA2MTEyNzIwMjM0MlqBDzIwMjYxMTI3MjA1MzQyWjAfBgNVHSME\n"
-"GDAWgBRokORnpKZTgMeGZqTx90tD+4S9bTAdBgNVHQ4EFgQUaJDkZ6SmU4DHhmak8fdLQ/uE\n"
-"vW0wHQYJKoZIhvZ9B0EABBAwDhsIVjcuMTo0LjADAgSQMA0GCSqGSIb3DQEBBQUAA4IBAQCT\n"
-"1DCw1wMgKtD5Y+iRDAUgqV8ZyntyTtSx29CW+1RaGSwMCPeyvIWonX9tO1KzKtvn1ISMY/YP\n"
-"yyYBkVBs9F8U4pN0wBOeMDpQ47RgxRzwIkSNcUesyBrJ6ZuaAGAT/3B+XxFNSRuzFVJ7yVTa\n"
-"v52Vr2ua2J7p8eRDjeIRRDq/r72DQnNSi6q7pynP9WQcCk3RvKqsnyrQ/39/2n3qse0wJcGE\n"
-"2jTSW3iDVuycNsMm4hH2Z0kdkquM++v/eu6FSqdQgPCnXEqULl8FmTxSQeDNtGPPAUO6nIPc\n"
-"j2A781q0tHuu2guQOHXvgR1m0vdXcDazv/wor3ElhVsT/h5/WrQ8\n"
-"-----END CERTIFICATE-----",
-
 /* COMODO ECC Certification Authority */
 "-----BEGIN CERTIFICATE-----\n"
 "MIICiTCCAg+gAwIBAgIQH0evqmIAcFBUTAGem2OZKjAKBggqhkjOPQQDAzCBhTELMAkGA1UE\n"
@@ -2641,27 +2615,6 @@
 "Jsas7a5wFsWqynKKTbDPAYsDP27X\n"
 "-----END CERTIFICATE-----",
 
-/* SecureSign Root CA12 */
-"-----BEGIN CERTIFICATE-----\n"
-"MIIDcjCCAlqgAwIBAgIUZvnHwa/swlG07VOX5uaCwysckBYwDQYJKoZIhvcNAQELBQAwUTEL\n"
-"MAkGA1UEBhMCSlAxIzAhBgNVBAoTGkN5YmVydHJ1c3QgSmFwYW4gQ28uLCBMdGQuMR0wGwYD\n"
-"VQQDExRTZWN1cmVTaWduIFJvb3QgQ0ExMjAeFw0yMDA0MDgwNTM2NDZaFw00MDA0MDgwNTM2\n"
-"NDZaMFExCzAJBgNVBAYTAkpQMSMwIQYDVQQKExpDeWJlcnRydXN0IEphcGFuIENvLiwgTHRk\n"
-"LjEdMBsGA1UEAxMUU2VjdXJlU2lnbiBSb290IENBMTIwggEiMA0GCSqGSIb3DQEBAQUAA4IB\n"
-"DwAwggEKAoIBAQC6OcE3emhFKxS06+QT61d1I02PJC0W6K6OyX2kVzsqdiUzg2zqMoqUm048\n"
-"luT9Ub+ZyZN+v/mtp7JIKwccJ/VMvHASd6SFVLX9kHrko+RRWAPNEHl57muTH2SOa2SroxPj\n"
-"cf59q5zdJ1M3s6oYwlkm7Fsf0uZlfO+TvdhYXAvA42VvPMfKWeP+bl+sg779XSVOKik71gur\n"
-"FzJ4pOE+lEa+Ym6b3kaosRbnhW70CEBFEaCeVESE99g2zvVQR9wsMJvuwPWW0v4JhscGWa5P\n"
-"ro4RmHvzC1KqYiaqId+OJTN5lxZJjfU+1UefNzFJM3IFTQy2VYzxV4+Kh9GtxRESOaCtAgMB\n"
-"AAGjQjBAMA8GA1UdEwEB/wQFMAMBAf8wDgYDVR0PAQH/BAQDAgEGMB0GA1UdDgQWBBRXNPN0\n"
-"zwRL1SXm8UC2LEzZLemgrTANBgkqhkiG9w0BAQsFAAOCAQEAPrvbFxbS8hQBICw4g0utvsqF\n"
-"epq2m2um4fylOqyttCg6r9cBg0krY6LdmmQOmFxv3Y67ilQiLUoT865AQ9tPkbeGGuwAtEGB\n"
-"pE/6aouIs3YIcipJQMPTw4WJmBClnW8Zt7vPemVV2zfrPIpyMpcemik+rY3moxtt9XUa5rBo\n"
-"uVui7mlHJzWhhpmA8zNL4WukJsPvdFlseqJkth5Ew1DgDzk9qTPxpfPSvWKErI4cqc1avTc7\n"
-"bgoitPQV55FYxTpE05Uo2cBl6XLK0A+9H7MV2anjpEcJnuDLN/v9vZfVvhgaaaI5gdka9at/\n"
-"yOPiZwud9AzqVN/Ssq+xIvEg37xEHA==\n"
-"-----END CERTIFICATE-----",
-
 /* SecureSign Root CA14 */
 "-----BEGIN CERTIFICATE-----\n"
 "MIIFcjCCA1qgAwIBAgIUZNtaDCBO6Ncpd8hQJ6JaJ90t8sswDQYJKoZIhvcNAQEMBQAwUTEL\n"
diff --git a/tools/certdata.txt b/tools/certdata.txt
index 97b118f687..fafc33ddee 100644
--- a/tools/certdata.txt
+++ b/tools/certdata.txt
@@ -371,179 +371,6 @@ CKA_TRUST_EMAIL_PROTECTION CK_TRUST CKT_NSS_TRUSTED_DELEGATOR
 CKA_TRUST_CODE_SIGNING CK_TRUST CKT_NSS_MUST_VERIFY_TRUST
 CKA_TRUST_STEP_UP_APPROVED CK_BBOOL CK_FALSE
 
-#
-# Certificate "Entrust Root Certification Authority"
-#
-# Issuer: CN=Entrust Root Certification Authority,OU="(c) 2006 Entrust, Inc.",OU=www.entrust.net/CPS is incorporated by reference,O="Entrust, Inc.",C=US
-# Serial Number: 1164660820 (0x456b5054)
-# Subject: CN=Entrust Root Certification Authority,OU="(c) 2006 Entrust, Inc.",OU=www.entrust.net/CPS is incorporated by reference,O="Entrust, Inc.",C=US
-# Not Valid Before: Mon Nov 27 20:23:42 2006
-# Not Valid After : Fri Nov 27 20:53:42 2026
-# Fingerprint (SHA-256): 73:C1:76:43:4F:1B:C6:D5:AD:F4:5B:0E:76:E7:27:28:7C:8D:E5:76:16:C1:E6:E6:14:1A:2B:2C:BC:7D:8E:4C
-# Fingerprint (SHA1): B3:1E:B1:B7:40:E3:6C:84:02:DA:DC:37:D4:4D:F5:D4:67:49:52:F9
-CKA_CLASS CK_OBJECT_CLASS CKO_CERTIFICATE
-CKA_TOKEN CK_BBOOL CK_TRUE
-CKA_PRIVATE CK_BBOOL CK_FALSE
-CKA_MODIFIABLE CK_BBOOL CK_FALSE
-CKA_LABEL UTF8 "Entrust Root Certification Authority"
-CKA_CERTIFICATE_TYPE CK_CERTIFICATE_TYPE CKC_X_509
-CKA_SUBJECT MULTILINE_OCTAL
-\060\201\260\061\013\060\011\006\003\125\004\006\023\002\125\123
-\061\026\060\024\006\003\125\004\012\023\015\105\156\164\162\165
-\163\164\054\040\111\156\143\056\061\071\060\067\006\003\125\004
-\013\023\060\167\167\167\056\145\156\164\162\165\163\164\056\156
-\145\164\057\103\120\123\040\151\163\040\151\156\143\157\162\160
-\157\162\141\164\145\144\040\142\171\040\162\145\146\145\162\145
-\156\143\145\061\037\060\035\006\003\125\004\013\023\026\050\143
-\051\040\062\060\060\066\040\105\156\164\162\165\163\164\054\040
-\111\156\143\056\061\055\060\053\006\003\125\004\003\023\044\105
-\156\164\162\165\163\164\040\122\157\157\164\040\103\145\162\164
-\151\146\151\143\141\164\151\157\156\040\101\165\164\150\157\162
-\151\164\171
-END
-CKA_ID UTF8 "0"
-CKA_ISSUER MULTILINE_OCTAL
-\060\201\260\061\013\060\011\006\003\125\004\006\023\002\125\123
-\061\026\060\024\006\003\125\004\012\023\015\105\156\164\162\165
-\163\164\054\040\111\156\143\056\061\071\060\067\006\003\125\004
-\013\023\060\167\167\167\056\145\156\164\162\165\163\164\056\156
-\145\164\057\103\120\123\040\151\163\040\151\156\143\157\162\160
-\157\162\141\164\145\144\040\142\171\040\162\145\146\145\162\145
-\156\143\145\061\037\060\035\006\003\125\004\013\023\026\050\143
-\051\040\062\060\060\066\040\105\156\164\162\165\163\164\054\040
-\111\156\143\056\061\055\060\053\006\003\125\004\003\023\044\105
-\156\164\162\165\163\164\040\122\157\157\164\040\103\145\162\164
-\151\146\151\143\141\164\151\157\156\040\101\165\164\150\157\162
-\151\164\171
-END
-CKA_SERIAL_NUMBER MULTILINE_OCTAL
-\002\004\105\153\120\124
-END
-CKA_VALUE MULTILINE_OCTAL
-\060\202\004\221\060\202\003\171\240\003\002\001\002\002\004\105
-\153\120\124\060\015\006\011\052\206\110\206\367\015\001\001\005
-\005\000\060\201\260\061\013\060\011\006\003\125\004\006\023\002
-\125\123\061\026\060\024\006\003\125\004\012\023\015\105\156\164
-\162\165\163\164\054\040\111\156\143\056\061\071\060\067\006\003
-\125\004\013\023\060\167\167\167\056\145\156\164\162\165\163\164
-\056\156\145\164\057\103\120\123\040\151\163\040\151\156\143\157
-\162\160\157\162\141\164\145\144\040\142\171\040\162\145\146\145
-\162\145\156\143\145\061\037\060\035\006\003\125\004\013\023\026
-\050\143\051\040\062\060\060\066\040\105\156\164\162\165\163\164
-\054\040\111\156\143\056\061\055\060\053\006\003\125\004\003\023
-\044\105\156\164\162\165\163\164\040\122\157\157\164\040\103\145
-\162\164\151\146\151\143\141\164\151\157\156\040\101\165\164\150
-\157\162\151\164\171\060\036\027\015\060\066\061\061\062\067\062
-\060\062\063\064\062\132\027\015\062\066\061\061\062\067\062\060
-\065\063\064\062\132\060\201\260\061\013\060\011\006\003\125\004
-\006\023\002\125\123\061\026\060\024\006\003\125\004\012\023\015
-\105\156\164\162\165\163\164\054\040\111\156\143\056\061\071\060
-\067\006\003\125\004\013\023\060\167\167\167\056\145\156\164\162
-\165\163\164\056\156\145\164\057\103\120\123\040\151\163\040\151
-\156\143\157\162\160\157\162\141\164\145\144\040\142\171\040\162
-\145\146\145\162\145\156\143\145\061\037\060\035\006\003\125\004
-\013\023\026\050\143\051\040\062\060\060\066\040\105\156\164\162
-\165\163\164\054\040\111\156\143\056\061\055\060\053\006\003\125
-\004\003\023\044\105\156\164\162\165\163\164\040\122\157\157\164
-\040\103\145\162\164\151\146\151\143\141\164\151\157\156\040\101
-\165\164\150\157\162\151\164\171\060\202\001\042\060\015\006\011
-\052\206\110\206\367\015\001\001\001\005\000\003\202\001\017\000
-\060\202\001\012\002\202\001\001\000\266\225\266\103\102\372\306
-\155\052\157\110\337\224\114\071\127\005\356\303\171\021\101\150
-\066\355\354\376\232\001\217\241\070\050\374\367\020\106\146\056
-\115\036\032\261\032\116\306\321\300\225\210\260\311\377\061\213
-\063\003\333\267\203\173\076\040\204\136\355\262\126\050\247\370
-\340\271\100\161\067\305\313\107\016\227\052\150\300\042\225\142
-\025\333\107\331\365\320\053\377\202\113\311\255\076\336\114\333
-\220\200\120\077\011\212\204\000\354\060\012\075\030\315\373\375
-\052\131\232\043\225\027\054\105\236\037\156\103\171\155\014\134
-\230\376\110\247\305\043\107\134\136\375\156\347\036\264\366\150
-\105\321\206\203\133\242\212\215\261\343\051\200\376\045\161\210
-\255\276\274\217\254\122\226\113\252\121\215\344\023\061\031\350
-\116\115\237\333\254\263\152\325\274\071\124\161\312\172\172\177
-\220\335\175\035\200\331\201\273\131\046\302\021\376\346\223\342
-\367\200\344\145\373\064\067\016\051\200\160\115\257\070\206\056
-\236\177\127\257\236\027\256\353\034\313\050\041\137\266\034\330
-\347\242\004\042\371\323\332\330\313\002\003\001\000\001\243\201
-\260\060\201\255\060\016\006\003\125\035\017\001\001\377\004\004
-\003\002\001\006\060\017\006\003\125\035\023\001\001\377\004\005
-\060\003\001\001\377\060\053\006\003\125\035\020\004\044\060\042
-\200\017\062\060\060\066\061\061\062\067\062\060\062\063\064\062
-\132\201\017\062\060\062\066\061\061\062\067\062\060\065\063\064
-\062\132\060\037\006\003\125\035\043\004\030\060\026\200\024\150
-\220\344\147\244\246\123\200\307\206\146\244\361\367\113\103\373
-\204\275\155\060\035\006\003\125\035\016\004\026\004\024\150\220
-\344\147\244\246\123\200\307\206\146\244\361\367\113\103\373\204
-\275\155\060\035\006\011\052\206\110\206\366\175\007\101\000\004
-\020\060\016\033\010\126\067\056\061\072\064\056\060\003\002\004
-\220\060\015\006\011\052\206\110\206\367\015\001\001\005\005\000
-\003\202\001\001\000\223\324\060\260\327\003\040\052\320\371\143
-\350\221\014\005\040\251\137\031\312\173\162\116\324\261\333\320
-\226\373\124\132\031\054\014\010\367\262\274\205\250\235\177\155
-\073\122\263\052\333\347\324\204\214\143\366\017\313\046\001\221
-\120\154\364\137\024\342\223\164\300\023\236\060\072\120\343\264
-\140\305\034\360\042\104\215\161\107\254\310\032\311\351\233\232
-\000\140\023\377\160\176\137\021\115\111\033\263\025\122\173\311
-\124\332\277\235\225\257\153\232\330\236\351\361\344\103\215\342
-\021\104\072\277\257\275\203\102\163\122\213\252\273\247\051\317
-\365\144\034\012\115\321\274\252\254\237\052\320\377\177\177\332
-\175\352\261\355\060\045\301\204\332\064\322\133\170\203\126\354
-\234\066\303\046\342\021\366\147\111\035\222\253\214\373\353\377
-\172\356\205\112\247\120\200\360\247\134\112\224\056\137\005\231
-\074\122\101\340\315\264\143\317\001\103\272\234\203\334\217\140
-\073\363\132\264\264\173\256\332\013\220\070\165\357\201\035\146
-\322\367\127\160\066\263\277\374\050\257\161\045\205\133\023\376
-\036\177\132\264\074
-END
-CKA_NSS_MOZILLA_CA_POLICY CK_BBOOL CK_TRUE
-# For Server Distrust After: Sat Nov 30 23:59:59 2024
-CKA_NSS_SERVER_DISTRUST_AFTER MULTILINE_OCTAL
-\062\064\061\061\063\060\062\063\065\071\065\071\132
-END
-CKA_NSS_EMAIL_DISTRUST_AFTER CK_BBOOL CK_FALSE
-
-# Trust for "Entrust Root Certification Authority"
-# Issuer: CN=Entrust Root Certification Authority,OU="(c) 2006 Entrust, Inc.",OU=www.entrust.net/CPS is incorporated by reference,O="Entrust, Inc.",C=US
-# Serial Number: 1164660820 (0x456b5054)
-# Subject: CN=Entrust Root Certification Authority,OU="(c) 2006 Entrust, Inc.",OU=www.entrust.net/CPS is incorporated by reference,O="Entrust, Inc.",C=US
-# Not Valid Before: Mon Nov 27 20:23:42 2006
-# Not Valid After : Fri Nov 27 20:53:42 2026
-# Fingerprint (SHA-256): 73:C1:76:43:4F:1B:C6:D5:AD:F4:5B:0E:76:E7:27:28:7C:8D:E5:76:16:C1:E6:E6:14:1A:2B:2C:BC:7D:8E:4C
-# Fingerprint (SHA1): B3:1E:B1:B7:40:E3:6C:84:02:DA:DC:37:D4:4D:F5:D4:67:49:52:F9
-CKA_CLASS CK_OBJECT_CLASS CKO_NSS_TRUST
-CKA_TOKEN CK_BBOOL CK_TRUE
-CKA_PRIVATE CK_BBOOL CK_FALSE
-CKA_MODIFIABLE CK_BBOOL CK_FALSE
-CKA_LABEL UTF8 "Entrust Root Certification Authority"
-CKA_CERT_SHA1_HASH MULTILINE_OCTAL
-\263\036\261\267\100\343\154\204\002\332\334\067\324\115\365\324
-\147\111\122\371
-END
-CKA_CERT_MD5_HASH MULTILINE_OCTAL
-\326\245\303\355\135\335\076\000\301\075\207\222\037\035\077\344
-END
-CKA_ISSUER MULTILINE_OCTAL
-\060\201\260\061\013\060\011\006\003\125\004\006\023\002\125\123
-\061\026\060\024\006\003\125\004\012\023\015\105\156\164\162\165
-\163\164\054\040\111\156\143\056\061\071\060\067\006\003\125\004
-\013\023\060\167\167\167\056\145\156\164\162\165\163\164\056\156
-\145\164\057\103\120\123\040\151\163\040\151\156\143\157\162\160
-\157\162\141\164\145\144\040\142\171\040\162\145\146\145\162\145
-\156\143\145\061\037\060\035\006\003\125\004\013\023\026\050\143
-\051\040\062\060\060\066\040\105\156\164\162\165\163\164\054\040
-\111\156\143\056\061\055\060\053\006\003\125\004\003\023\044\105
-\156\164\162\165\163\164\040\122\157\157\164\040\103\145\162\164
-\151\146\151\143\141\164\151\157\156\040\101\165\164\150\157\162
-\151\164\171
-END
-CKA_SERIAL_NUMBER MULTILINE_OCTAL
-\002\004\105\153\120\124
-END
-CKA_TRUST_SERVER_AUTH CK_TRUST CKT_NSS_TRUSTED_DELEGATOR
-CKA_TRUST_EMAIL_PROTECTION CK_TRUST CKT_NSS_MUST_VERIFY_TRUST
-CKA_TRUST_CODE_SIGNING CK_TRUST CKT_NSS_MUST_VERIFY_TRUST
-CKA_TRUST_STEP_UP_APPROVED CK_BBOOL CK_FALSE
-
 #
 # Certificate "Certum Root CA"
 #
@@ -21216,7 +21043,7 @@ CKA_SERIAL_NUMBER MULTILINE_OCTAL
 \261\211\270\161\223\017
 END
 CKA_TRUST_SERVER_AUTH CK_TRUST CKT_NSS_TRUSTED_DELEGATOR
-CKA_TRUST_EMAIL_PROTECTION CK_TRUST CKT_NSS_TRUSTED_DELEGATOR
+CKA_TRUST_EMAIL_PROTECTION CK_TRUST CKT_NSS_MUST_VERIFY_TRUST
 CKA_TRUST_CODE_SIGNING CK_TRUST CKT_NSS_MUST_VERIFY_TRUST
 CKA_TRUST_STEP_UP_APPROVED CK_BBOOL CK_FALSE
 
@@ -21334,7 +21161,7 @@ CKA_SERIAL_NUMBER MULTILINE_OCTAL
 \313\052\164\025\326\327
 END
 CKA_TRUST_SERVER_AUTH CK_TRUST CKT_NSS_TRUSTED_DELEGATOR
-CKA_TRUST_EMAIL_PROTECTION CK_TRUST CKT_NSS_TRUSTED_DELEGATOR
+CKA_TRUST_EMAIL_PROTECTION CK_TRUST CKT_NSS_MUST_VERIFY_TRUST
 CKA_TRUST_CODE_SIGNING CK_TRUST CKT_NSS_MUST_VERIFY_TRUST
 CKA_TRUST_STEP_UP_APPROVED CK_BBOOL CK_FALSE
 
@@ -22552,142 +22379,6 @@ CKA_TRUST_EMAIL_PROTECTION CK_TRUST CKT_NSS_TRUSTED_DELEGATOR
 CKA_TRUST_CODE_SIGNING CK_TRUST CKT_NSS_MUST_VERIFY_TRUST
 CKA_TRUST_STEP_UP_APPROVED CK_BBOOL CK_FALSE
 
-#
-# Certificate "SecureSign Root CA12"
-#
-# Issuer: CN=SecureSign Root CA12,O="Cybertrust Japan Co., Ltd.",C=JP
-# Serial Number:66:f9:c7:c1:af:ec:c2:51:b4:ed:53:97:e6:e6:82:c3:2b:1c:90:16
-# Subject: CN=SecureSign Root CA12,O="Cybertrust Japan Co., Ltd.",C=JP
-# Not Valid Before: Wed Apr 08 05:36:46 2020
-# Not Valid After : Sun Apr 08 05:36:46 2040
-# Fingerprint (SHA-256): 3F:03:4B:B5:70:4D:44:B2:D0:85:45:A0:20:57:DE:93:EB:F3:90:5F:CE:72:1A:CB:C7:30:C0:6D:DA:EE:90:4E
-# Fingerprint (SHA1): 7A:22:1E:3D:DE:1B:06:AC:9E:C8:47:70:16:8E:3C:E5:F7:6B:06:F4
-CKA_CLASS CK_OBJECT_CLASS CKO_CERTIFICATE
-CKA_TOKEN CK_BBOOL CK_TRUE
-CKA_PRIVATE CK_BBOOL CK_FALSE
-CKA_MODIFIABLE CK_BBOOL CK_FALSE
-CKA_LABEL UTF8 "SecureSign Root CA12"
-CKA_CERTIFICATE_TYPE CK_CERTIFICATE_TYPE CKC_X_509
-CKA_SUBJECT MULTILINE_OCTAL
-\060\121\061\013\060\011\006\003\125\004\006\023\002\112\120\061
-\043\060\041\006\003\125\004\012\023\032\103\171\142\145\162\164
-\162\165\163\164\040\112\141\160\141\156\040\103\157\056\054\040
-\114\164\144\056\061\035\060\033\006\003\125\004\003\023\024\123
-\145\143\165\162\145\123\151\147\156\040\122\157\157\164\040\103
-\101\061\062
-END
-CKA_ID UTF8 "0"
-CKA_ISSUER MULTILINE_OCTAL
-\060\121\061\013\060\011\006\003\125\004\006\023\002\112\120\061
-\043\060\041\006\003\125\004\012\023\032\103\171\142\145\162\164
-\162\165\163\164\040\112\141\160\141\156\040\103\157\056\054\040
-\114\164\144\056\061\035\060\033\006\003\125\004\003\023\024\123
-\145\143\165\162\145\123\151\147\156\040\122\157\157\164\040\103
-\101\061\062
-END
-CKA_SERIAL_NUMBER MULTILINE_OCTAL
-\002\024\146\371\307\301\257\354\302\121\264\355\123\227\346\346
-\202\303\053\034\220\026
-END
-CKA_VALUE MULTILINE_OCTAL
-\060\202\003\162\060\202\002\132\240\003\002\001\002\002\024\146
-\371\307\301\257\354\302\121\264\355\123\227\346\346\202\303\053
-\034\220\026\060\015\006\011\052\206\110\206\367\015\001\001\013
-\005\000\060\121\061\013\060\011\006\003\125\004\006\023\002\112
-\120\061\043\060\041\006\003\125\004\012\023\032\103\171\142\145
-\162\164\162\165\163\164\040\112\141\160\141\156\040\103\157\056
-\054\040\114\164\144\056\061\035\060\033\006\003\125\004\003\023
-\024\123\145\143\165\162\145\123\151\147\156\040\122\157\157\164
-\040\103\101\061\062\060\036\027\015\062\060\060\064\060\070\060
-\065\063\066\064\066\132\027\015\064\060\060\064\060\070\060\065
-\063\066\064\066\132\060\121\061\013\060\011\006\003\125\004\006
-\023\002\112\120\061\043\060\041\006\003\125\004\012\023\032\103
-\171\142\145\162\164\162\165\163\164\040\112\141\160\141\156\040
-\103\157\056\054\040\114\164\144\056\061\035\060\033\006\003\125
-\004\003\023\024\123\145\143\165\162\145\123\151\147\156\040\122
-\157\157\164\040\103\101\061\062\060\202\001\042\060\015\006\011
-\052\206\110\206\367\015\001\001\001\005\000\003\202\001\017\000
-\060\202\001\012\002\202\001\001\000\272\071\301\067\172\150\105
-\053\024\264\353\344\023\353\127\165\043\115\217\044\055\026\350
-\256\216\311\175\244\127\073\052\166\045\063\203\154\352\062\212
-\224\233\116\074\226\344\375\121\277\231\311\223\176\277\371\255
-\247\262\110\053\007\034\047\365\114\274\160\022\167\244\205\124
-\265\375\220\172\344\243\344\121\130\003\315\020\171\171\356\153
-\223\037\144\216\153\144\253\243\023\343\161\376\175\253\234\335
-\047\123\067\263\252\030\302\131\046\354\133\037\322\346\145\174
-\357\223\275\330\130\134\013\300\343\145\157\074\307\312\131\343
-\376\156\137\254\203\276\375\135\045\116\052\051\073\326\013\253
-\027\062\170\244\341\076\224\106\276\142\156\233\336\106\250\261
-\026\347\205\156\364\010\100\105\021\240\236\124\104\204\367\330
-\066\316\365\120\107\334\054\060\233\356\300\365\226\322\376\011
-\206\307\006\131\256\117\256\216\021\230\173\363\013\122\252\142
-\046\252\041\337\216\045\063\171\227\026\111\215\365\076\325\107
-\237\067\061\111\063\162\005\115\014\266\125\214\361\127\217\212
-\207\321\255\305\021\022\071\240\255\002\003\001\000\001\243\102
-\060\100\060\017\006\003\125\035\023\001\001\377\004\005\060\003
-\001\001\377\060\016\006\003\125\035\017\001\001\377\004\004\003
-\002\001\006\060\035\006\003\125\035\016\004\026\004\024\127\064
-\363\164\317\004\113\325\045\346\361\100\266\054\114\331\055\351
-\240\255\060\015\006\011\052\206\110\206\367\015\001\001\013\005
-\000\003\202\001\001\000\076\273\333\027\026\322\362\024\001\040
-\054\070\203\113\255\276\312\205\172\232\266\233\153\246\341\374
-\245\072\254\255\264\050\072\257\327\001\203\111\053\143\242\335
-\232\144\016\230\134\157\335\216\273\212\124\042\055\112\023\363
-\256\100\103\333\117\221\267\206\032\354\000\264\101\201\244\117
-\372\152\213\210\263\166\010\162\052\111\100\303\323\303\205\211
-\230\020\245\235\157\031\267\273\317\172\145\125\333\067\353\074
-\212\162\062\227\036\232\051\076\255\215\346\243\033\155\365\165
-\032\346\260\150\271\133\242\356\151\107\047\065\241\206\231\200
-\363\063\113\341\153\244\046\303\357\164\131\154\172\242\144\266
-\036\104\303\120\340\017\071\075\251\063\361\245\363\322\275\142
-\204\254\216\034\251\315\132\275\067\073\156\012\042\264\364\025
-\347\221\130\305\072\104\323\225\050\331\300\145\351\162\312\320
-\017\275\037\263\025\331\251\343\244\107\011\236\340\313\067\373
-\375\275\227\325\276\030\032\151\242\071\201\331\032\365\253\177
-\310\343\342\147\013\235\364\014\352\124\337\322\262\257\261\042
-\361\040\337\274\104\034
-END
-CKA_NSS_MOZILLA_CA_POLICY CK_BBOOL CK_TRUE
-CKA_NSS_SERVER_DISTRUST_AFTER CK_BBOOL CK_FALSE
-CKA_NSS_EMAIL_DISTRUST_AFTER CK_BBOOL CK_FALSE
-
-# Trust for "SecureSign Root CA12"
-# Issuer: CN=SecureSign Root CA12,O="Cybertrust Japan Co., Ltd.",C=JP
-# Serial Number:66:f9:c7:c1:af:ec:c2:51:b4:ed:53:97:e6:e6:82:c3:2b:1c:90:16
-# Subject: CN=SecureSign Root CA12,O="Cybertrust Japan Co., Ltd.",C=JP
-# Not Valid Before: Wed Apr 08 05:36:46 2020
-# Not Valid After : Sun Apr 08 05:36:46 2040
-# Fingerprint (SHA-256): 3F:03:4B:B5:70:4D:44:B2:D0:85:45:A0:20:57:DE:93:EB:F3:90:5F:CE:72:1A:CB:C7:30:C0:6D:DA:EE:90:4E
-# Fingerprint (SHA1): 7A:22:1E:3D:DE:1B:06:AC:9E:C8:47:70:16:8E:3C:E5:F7:6B:06:F4
-CKA_CLASS CK_OBJECT_CLASS CKO_NSS_TRUST
-CKA_TOKEN CK_BBOOL CK_TRUE
-CKA_PRIVATE CK_BBOOL CK_FALSE
-CKA_MODIFIABLE CK_BBOOL CK_FALSE
-CKA_LABEL UTF8 "SecureSign Root CA12"
-CKA_CERT_SHA1_HASH MULTILINE_OCTAL
-\172\042\036\075\336\033\006\254\236\310\107\160\026\216\074\345
-\367\153\006\364
-END
-CKA_CERT_MD5_HASH MULTILINE_OCTAL
-\306\211\312\144\102\233\142\010\111\013\036\177\351\007\075\350
-END
-CKA_ISSUER MULTILINE_OCTAL
-\060\121\061\013\060\011\006\003\125\004\006\023\002\112\120\061
-\043\060\041\006\003\125\004\012\023\032\103\171\142\145\162\164
-\162\165\163\164\040\112\141\160\141\156\040\103\157\056\054\040
-\114\164\144\056\061\035\060\033\006\003\125\004\003\023\024\123
-\145\143\165\162\145\123\151\147\156\040\122\157\157\164\040\103
-\101\061\062
-END
-CKA_SERIAL_NUMBER MULTILINE_OCTAL
-\002\024\146\371\307\301\257\354\302\121\264\355\123\227\346\346
-\202\303\053\034\220\026
-END
-CKA_TRUST_SERVER_AUTH CK_TRUST CKT_NSS_TRUSTED_DELEGATOR
-CKA_TRUST_EMAIL_PROTECTION CK_TRUST CKT_NSS_MUST_VERIFY_TRUST
-CKA_TRUST_CODE_SIGNING CK_TRUST CKT_NSS_MUST_VERIFY_TRUST
-CKA_TRUST_STEP_UP_APPROVED CK_BBOOL CK_FALSE
-
 #
 # Certificate "SecureSign Root CA14"
 #
@@ -24904,3 +24595,171 @@ CKA_TRUST_SERVER_AUTH CK_TRUST CKT_NSS_TRUSTED_DELEGATOR
 CKA_TRUST_EMAIL_PROTECTION CK_TRUST CKT_NSS_MUST_VERIFY_TRUST
 CKA_TRUST_CODE_SIGNING CK_TRUST CKT_NSS_MUST_VERIFY_TRUST
 CKA_TRUST_STEP_UP_APPROVED CK_BBOOL CK_FALSE
+
+#
+# Certificate "SecureSign Root CA16"
+#
+# Issuer: CN=SecureSign Root CA16,O="Cybertrust Japan Co., Ltd.",C=JP
+# Serial Number:54:7b:8d:ab:53:11:00:77:a8:18:03:ae:a1:2b:11:29:ab:42:e0:45
+# Subject: CN=SecureSign Root CA16,O="Cybertrust Japan Co., Ltd.",C=JP
+# Not Valid Before: Tue Jul 30 07:08:11 2024
+# Not Valid After : Fri Jul 29 06:55:40 2044
+# Fingerprint (SHA-256): 4C:1C:CD:24:F1:7E:95:0F:C1:85:36:B3:3C:AF:E3:22:93:CF:C3:3E:84:67:B4:1E:1C:69:30:55:D7:F5:13:BF
+# Fingerprint (SHA1): D1:79:17:EC:45:E2:A0:CA:D7:74:51:30:10:A4:C6:5C:AA:B3:3C:49
+CKA_CLASS CK_OBJECT_CLASS CKO_CERTIFICATE
+CKA_TOKEN CK_BBOOL CK_TRUE
+CKA_PRIVATE CK_BBOOL CK_FALSE
+CKA_MODIFIABLE CK_BBOOL CK_FALSE
+CKA_LABEL UTF8 "SecureSign Root CA16"
+CKA_CERTIFICATE_TYPE CK_CERTIFICATE_TYPE CKC_X_509
+CKA_SUBJECT MULTILINE_OCTAL
+\060\121\061\013\060\011\006\003\125\004\006\023\002\112\120\061
+\043\060\041\006\003\125\004\012\023\032\103\171\142\145\162\164
+\162\165\163\164\040\112\141\160\141\156\040\103\157\056\054\040
+\114\164\144\056\061\035\060\033\006\003\125\004\003\023\024\123
+\145\143\165\162\145\123\151\147\156\040\122\157\157\164\040\103
+\101\061\066
+END
+CKA_ID UTF8 "0"
+CKA_ISSUER MULTILINE_OCTAL
+\060\121\061\013\060\011\006\003\125\004\006\023\002\112\120\061
+\043\060\041\006\003\125\004\012\023\032\103\171\142\145\162\164
+\162\165\163\164\040\112\141\160\141\156\040\103\157\056\054\040
+\114\164\144\056\061\035\060\033\006\003\125\004\003\023\024\123
+\145\143\165\162\145\123\151\147\156\040\122\157\157\164\040\103
+\101\061\066
+END
+CKA_SERIAL_NUMBER MULTILINE_OCTAL
+\002\024\124\173\215\253\123\021\000\167\250\030\003\256\241\053
+\021\051\253\102\340\105
+END
+CKA_VALUE MULTILINE_OCTAL
+\060\202\005\162\060\202\003\132\240\003\002\001\002\002\024\124
+\173\215\253\123\021\000\167\250\030\003\256\241\053\021\051\253
+\102\340\105\060\015\006\011\052\206\110\206\367\015\001\001\014
+\005\000\060\121\061\013\060\011\006\003\125\004\006\023\002\112
+\120\061\043\060\041\006\003\125\004\012\023\032\103\171\142\145
+\162\164\162\165\163\164\040\112\141\160\141\156\040\103\157\056
+\054\040\114\164\144\056\061\035\060\033\006\003\125\004\003\023
+\024\123\145\143\165\162\145\123\151\147\156\040\122\157\157\164
+\040\103\101\061\066\060\036\027\015\062\064\060\067\063\060\060
+\067\060\070\061\061\132\027\015\064\064\060\067\062\071\060\066
+\065\065\064\060\132\060\121\061\013\060\011\006\003\125\004\006
+\023\002\112\120\061\043\060\041\006\003\125\004\012\023\032\103
+\171\142\145\162\164\162\165\163\164\040\112\141\160\141\156\040
+\103\157\056\054\040\114\164\144\056\061\035\060\033\006\003\125
+\004\003\023\024\123\145\143\165\162\145\123\151\147\156\040\122
+\157\157\164\040\103\101\061\066\060\202\002\042\060\015\006\011
+\052\206\110\206\367\015\001\001\001\005\000\003\202\002\017\000
+\060\202\002\012\002\202\002\001\000\307\045\037\360\022\266\217
+\261\357\177\076\127\276\107\206\363\342\046\314\016\172\356\163
+\302\366\063\044\352\302\037\156\345\321\232\164\205\224\354\163
+\021\340\141\210\355\133\055\245\327\123\323\266\200\254\260\345
+\126\351\126\152\152\232\215\273\352\335\155\302\164\175\136\365
+\047\333\044\267\125\212\222\133\303\245\020\371\167\103\156\151
+\366\263\346\336\175\165\221\260\215\351\373\140\046\155\011\355
+\376\341\124\331\000\211\271\262\326\205\163\070\046\044\323\111
+\231\306\154\065\366\352\076\057\005\174\052\071\002\351\205\065
+\256\123\052\175\144\147\322\200\256\224\260\243\162\010\064\263
+\143\125\317\230\137\103\062\151\376\253\314\205\213\217\043\343
+\214\207\224\076\236\315\341\213\006\372\171\107\025\251\333\126
+\053\122\363\265\101\063\372\244\334\216\325\255\313\116\226\273
+\261\316\324\253\123\130\274\110\134\264\217\327\146\036\024\041
+\014\065\140\320\164\263\141\220\056\170\272\005\356\120\325\117
+\243\302\360\132\072\256\131\314\175\103\207\240\106\161\154\113
+\216\354\064\200\376\273\164\334\370\273\040\356\242\343\352\352
+\246\230\015\262\334\021\044\141\052\260\142\214\346\144\200\130
+\376\270\034\261\325\223\134\001\356\064\160\363\267\035\324\245
+\146\323\016\131\246\275\004\203\313\321\114\014\246\132\142\040
+\113\204\245\116\073\324\250\362\245\142\143\313\136\265\362\170
+\066\027\026\036\362\073\100\006\136\112\155\336\365\364\255\154
+\214\350\211\157\154\050\060\102\055\027\166\000\145\043\235\324
+\321\350\007\077\041\012\356\266\356\123\265\372\173\355\076\072
+\101\224\155\060\051\271\254\117\357\123\013\005\136\304\236\154
+\000\332\163\350\176\056\353\373\302\270\255\105\120\146\316\115
+\370\116\355\110\275\236\027\120\162\217\263\226\106\037\332\320
+\327\267\072\314\024\134\274\211\263\133\354\214\175\254\242\217
+\233\131\206\016\335\060\304\063\150\010\233\275\111\361\305\375
+\072\166\021\233\327\017\240\100\140\046\342\014\257\211\340\162
+\123\252\203\355\311\073\032\341\117\005\216\070\215\122\201\000
+\075\141\105\021\277\263\254\273\114\263\125\163\245\260\135\034
+\164\137\026\141\051\236\250\002\073\002\003\001\000\001\243\102
+\060\100\060\017\006\003\125\035\023\001\001\377\004\005\060\003
+\001\001\377\060\016\006\003\125\035\017\001\001\377\004\004\003
+\002\001\006\060\035\006\003\125\035\016\004\026\004\024\030\156
+\064\266\333\231\125\144\110\245\206\111\270\236\113\223\367\016
+\053\017\060\015\006\011\052\206\110\206\367\015\001\001\014\005
+\000\003\202\002\001\000\253\202\224\212\011\027\240\274\203\130
+\204\261\263\200\143\044\112\341\121\050\076\370\114\314\140\037
+\341\151\320\203\302\361\356\375\010\326\067\323\026\167\102\273
+\331\250\006\166\160\024\271\271\323\151\353\223\352\342\140\061
+\225\300\141\021\165\353\142\336\201\275\345\012\244\244\247\105
+\053\355\222\340\247\110\273\226\110\056\315\317\066\007\161\270
+\166\132\204\377\321\316\336\151\217\124\111\365\135\223\111\100
+\020\160\152\122\304\264\053\340\002\157\066\317\240\006\137\047
+\067\316\025\356\022\002\000\115\135\234\127\331\267\335\150\246
+\177\034\266\244\251\212\025\377\113\215\301\274\027\034\101\327
+\156\272\124\045\234\055\263\070\161\225\150\027\221\334\214\360
+\130\110\064\036\157\217\345\351\055\101\317\017\222\200\330\314
+\144\252\231\021\176\172\371\062\376\033\042\266\372\055\253\054
+\100\322\374\254\045\265\016\114\016\074\273\050\010\243\223\313
+\332\004\176\260\350\065\312\136\025\106\162\073\333\051\316\255
+\365\157\007\173\232\311\111\054\114\012\022\143\302\256\352\355
+\005\044\317\074\316\146\051\037\327\072\140\144\067\200\052\331
+\173\063\157\363\347\213\256\241\024\316\231\031\345\013\172\274
+\326\137\322\223\021\272\365\207\231\337\041\113\342\117\150\126
+\375\261\165\053\076\113\075\116\173\142\346\131\250\365\044\062
+\311\054\243\356\035\266\365\176\056\055\047\122\011\165\203\260
+\306\106\033\073\265\074\252\140\211\345\315\227\037\261\147\313
+\161\217\144\021\025\352\202\000\343\342\115\326\311\270\225\222
+\105\011\025\236\133\121\343\330\301\015\242\202\232\244\135\151
+\334\105\170\201\127\064\035\264\270\003\061\370\171\250\221\104
+\162\216\343\025\312\001\236\150\370\247\306\104\010\256\164\052
+\226\361\023\141\103\030\321\312\107\227\270\330\270\071\254\111
+\031\320\145\360\241\273\132\200\207\251\142\151\276\053\271\257
+\237\310\001\375\326\373\203\237\214\274\227\052\124\141\222\007
+\310\301\316\256\111\140\056\337\345\053\116\000\072\374\271\106
+\253\164\031\070\145\231\301\367\002\065\122\356\351\335\365\164
+\115\240\160\237\156\266\334\024\013\103\062\330\001\021\007\074
+\215\105\157\101\254\001
+END
+CKA_NSS_MOZILLA_CA_POLICY CK_BBOOL CK_TRUE
+CKA_NSS_SERVER_DISTRUST_AFTER CK_BBOOL CK_FALSE
+CKA_NSS_EMAIL_DISTRUST_AFTER CK_BBOOL CK_FALSE
+
+# Trust for "SecureSign Root CA16"
+# Issuer: CN=SecureSign Root CA16,O="Cybertrust Japan Co., Ltd.",C=JP
+# Serial Number:54:7b:8d:ab:53:11:00:77:a8:18:03:ae:a1:2b:11:29:ab:42:e0:45
+# Subject: CN=SecureSign Root CA16,O="Cybertrust Japan Co., Ltd.",C=JP
+# Not Valid Before: Tue Jul 30 07:08:11 2024
+# Not Valid After : Fri Jul 29 06:55:40 2044
+# Fingerprint (SHA-256): 4C:1C:CD:24:F1:7E:95:0F:C1:85:36:B3:3C:AF:E3:22:93:CF:C3:3E:84:67:B4:1E:1C:69:30:55:D7:F5:13:BF
+# Fingerprint (SHA1): D1:79:17:EC:45:E2:A0:CA:D7:74:51:30:10:A4:C6:5C:AA:B3:3C:49
+CKA_CLASS CK_OBJECT_CLASS CKO_NSS_TRUST
+CKA_TOKEN CK_BBOOL CK_TRUE
+CKA_PRIVATE CK_BBOOL CK_FALSE
+CKA_MODIFIABLE CK_BBOOL CK_FALSE
+CKA_LABEL UTF8 "SecureSign Root CA16"
+CKA_CERT_SHA1_HASH MULTILINE_OCTAL
+\321\171\027\354\105\342\240\312\327\164\121\060\020\244\306\134
+\252\263\074\111
+END
+CKA_CERT_MD5_HASH MULTILINE_OCTAL
+\150\132\170\276\353\165\351\257\156\376\274\220\301\201\062\245
+END
+CKA_ISSUER MULTILINE_OCTAL
+\060\121\061\013\060\011\006\003\125\004\006\023\002\112\120\061
+\043\060\041\006\003\125\004\012\023\032\103\171\142\145\162\164
+\162\165\163\164\040\112\141\160\141\156\040\103\157\056\054\040
+\114\164\144\056\061\035\060\033\006\003\125\004\003\023\024\123
+\145\143\165\162\145\123\151\147\156\040\122\157\157\164\040\103
+\101\061\066
+END
+CKA_SERIAL_NUMBER MULTILINE_OCTAL
+\002\024\124\173\215\253\123\021\000\167\250\030\003\256\241\053
+\021\051\253\102\340\105
+END
+CKA_TRUST_SERVER_AUTH CK_TRUST CKT_NSS_MUST_VERIFY_TRUST
+CKA_TRUST_EMAIL_PROTECTION CK_TRUST CKT_NSS_TRUSTED_DELEGATOR
+CKA_TRUST_CODE_SIGNING CK_TRUST CKT_NSS_MUST_VERIFY_TRUST
+CKA_TRUST_STEP_UP_APPROVED CK_BBOOL CK_FALSE
